### ISSUE TITLE
Opening an article in the browser causes a never-ending loading icon over the share icon

### ISSUE THREAD

**Issue by PopeRigby** [jangernert/FeedReader#971]
# Problem
Opening an article in the browser causes a never-ending loading icon over the share icon

# Steps to reproduce
1. Click on the share icon
2. Click on "Open in browser"

# Expected behavior
There is no loading icon over the share button

# Actual behavior
There's a loading icon over the share button

# Extra information
[Video of it happening](https://pixelfed.social/p/PopeRigby/115292215013675008)

**Comment by tanevanwifferen** [comment]
https://github.com/jangernert/FeedReader/blob/6905f466acb4b0fefd7f6269940202ff389a52dc/src/Widgets/SharePopover.vala#L139

needs a "shareDone" call. I can't edit, as github says I need to be on a branch or so, but forking and creating a PR is too much of a hassle for 1 single line for me, so I hope this is enough for you @jangernert

**Comment by PopeRigby** [comment]
Like this?
```
private void shareURL(string id, string url)
	{
		this.hide();
		startShare();
		shareInternal(id, url);
		string idString = (id == null || id == "") ? "" : @" to $id";
		Logger.debug(@"bookmark: $url$idString");
		shareDone();
	}
```

**Comment by tanevanwifferen** [comment]
Yeah, that is one change. When compiling I have found another issue on src/Backend/FeedServer.vala : 57, you should modify Secret.CollectionCreateFlags.COLLECTION_CREATE_NONE to be Secret.CollectionCreateFlags.NONE. Also, if you're on arch like me, and are using the aur package, you might need to remove the database (~/.local/share/feedreader/feedreader-7.db) before it works again...

**Comment by PopeRigby** [comment]
I'm having some dependency issues trying to build it on Arch. It says I'm missing `libpeas-1.0` but I have that installed. How did you get it to work?

**Comment by tanevanwifferen** [comment]
No clue? I didn't have any issues with that. Maybe libpeas-devel? or you might see some missing packages when running meson. I just remember I also had to change the json lib reference from jsoncpp to just json. Maybe your system has an issue like it?

**Comment by PopeRigby** [comment]
> Maybe libpeas-devel?

It doesn't look like that's in the Arch repos.

> or you might see some missing packages when running meson

That's what's happening. It's saying
```
Run-time dependency libpeas-1.0 found: NO (tried pkgconfig and cmake)

meson.build:51:0: ERROR: Dependency "libpeas-1.0" not found, tried pkgconfig and cmake
```

I checked, and my version of libpeas is `1.24.1-2`

**Comment by jangernert** [comment]
You definitely need the devel package of libpeas. How that is called in arch linux, I have no idea.

**Comment by PopeRigby** [comment]
After finding [this](https://github.com/TingPing/irc-client/issues/2) thread, I figured out I needed `gobject-introspection`. I was able to build it, but the loading icon problem persists.

**Comment by jangernert** [comment]
Did you actually quit the running instance of the application before starting the patched build?

**Comment by PopeRigby** [comment]
Oops. That's a dumb mistake to have made. Ok, so that fixed it. I'll submit a PR.
